src/pudl/scripts/dbt_helper.py

```diff
@@ -281,13 +284,18 @@ def _calculate_row_counts(
 
     row_count_query = f"""
 SELECT
-    CAST({partition_expr_sql} AS VARCHAR) AS partition,
+    CAST(COALESCE(CAST({partition_expr_sql} AS VARCHAR), '') AS VARCHAR) AS partition,
     COUNT(*) AS row_count
 FROM '{table_path}' {group_by_clause}
     """  # noqa: S608
 
-    new_row_counts = duckdb.sql(row_count_query).df().astype({"partition": str})
-    new_row_counts["table_name"] = table_name
+    new_row_counts = (
+        duckdb.sql(row_count_query)
+        .df()
+        .assign(table_name=table_name)
+        .astype({"partition": "string", "table_name": "string"})
+        .loc[:, ["table_name", "partition", "row_count"]]
+    )
```

Comment:
Munge the results here to ensure that they are comparable to the old row counts.